### PR TITLE
feat(ivy): support animation @triggers in templates

### PR DESCRIPTION
**Pull request by matsko**

### CODE REVIEW

File: packages/compiler-cli/test/compliance/r3_view_compiler_styling_spec.ts [hunk 1]
```diff
@@ -100,6 +100,143 @@ describe('compiler compliance: styling', () => {
...
+
+                @NgModule({declarations: [MyComponent]})
+                export class MyModule {}
+            `
+        }
+      };
+
+      const template = `
+        MyComponent.ngComponentDef = $r3$.ɵdefineComponent({
+          type: MyComponent,
```
Comment (mhevery) [inline packages/compiler-cli/test/compliance/r3_view_compiler_styling_spec.ts]: you can use `...` to elide irrelevant text and only keep animations.
```
MyComponent.ngComponentDef = $r3$.ɵdefineComponent({
 ...
  animations: [{name: "foo123"}, {name: "trigger123"}]
 ...
});
```

File: packages/compiler-cli/test/compliance/r3_view_compiler_styling_spec.ts [hunk 2]
```diff
@@ -100,6 +100,143 @@ describe('compiler compliance: styling', () => {
...
+          selectors:[["my-component"]],
+          factory:function MyComponent_Factory(t){
+            return new (t || MyComponent)();
+          },
+          features: [$r3$.ɵPublicFeature],
+          consts: 0,
+          vars: 0,
+          template:  function MyComponent_Template(rf, $ctx$) {
```
Comment (mhevery) [inline packages/compiler-cli/test/compliance/r3_view_compiler_styling_spec.ts]: same here, elide irrelevant text.

File: packages/compiler-cli/test/compliance/r3_view_compiler_styling_spec.ts [hunk 3]
```diff
@@ -180,6 +180,61 @@ describe('compiler compliance: styling', () => {
...
+            if (rf & 1) {
+              $r3$.ɵelement(0, "div", $e0_attrs$);
+              $r3$.ɵelement(1, "div", $e1_attrs$);
+              $r3$.ɵelement(2, "div", $e2_attrs$);
+            }
+            if (rf & 2) {
+              $r3$.ɵelementAttribute(0, "@foo", $r3$.ɵbind(ctx.exp));
+              $r3$.ɵelementAttribute(1, "@bar", undefined);
+              $r3$.ɵelementAttribute(2, "@baz", undefined);
```
[the remark this answers is not in the document]
Comment (mhevery) [reply packages/compiler-cli/test/compliance/r3_view_compiler_styling_spec.ts]: Yes, this test is not canonical, and will confuse people reading it. Can you make it canonical by binding to a value, which may be constant for the test.

**Review by mhevery — APPROVED** [review]
Conditionally on the clean up comments.

File: packages/compiler-cli/test/compliance/r3_view_compiler_styling_spec.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (matsko) [reply packages/compiler-cli/test/compliance/r3_view_compiler_styling_spec.ts]: How's about just setting the `undefined` value to be `NO_CHANGE`?

[the remark this answers is not in the document]
Comment (JoostK) [reply packages/compiler-cli/test/compliance/r3_view_compiler_styling_spec.ts]: @matsko It won't do a thing, then ;) Basically, constant attributes are usually applied for `RenderFlags.Create` by passing the constant attributes as third argument to `elementStart`.

[the remark this answers is not in the document]
Comment (matsko) [reply packages/compiler-cli/test/compliance/r3_view_compiler_styling_spec.ts]: OK the instructions are gone.

**Pull request closed without merging**